src/DataProvider/OperationDataProviderTrait.php

```diff
@@ -75,7 +76,18 @@ private function getSubresourceData($identifiers, array $attributes, array $cont
             throw new RuntimeException('Subresources not supported');
         }
 
-        return $this->subresourceDataProvider->getSubresource($attributes['resource_class'], $identifiers, $attributes['subresource_context'] + $context, $attributes['subresource_operation_name']);
+        $subresourceIdentifiers = [];
+        foreach ($attributes['subresource_context']['identifiers'] as list($identifier, , $isItem, $identifiedBy)) {
+            if ($isItem) {
+                $subresourceIdentifiers[$identifier] = [$identifiedBy[0] => current($identifiers)];
+                next($identifiers);
```

Comment:
Shouldn't we use an index instead?

Reply:
this will be simplified by #3689 it's more of a workaround for now.